src/ORM/Rule/ExistsIn.php

```diff
@@ -98,6 +109,11 @@ public function __invoke(EntityInterface $entity, array $options)
             return true;
         }
 
+        if ($this->_options['allowSqlNulls'] === true
+            && $this->_checkPartialSchemaNulls($entity, $source) === true
+        ) {
+            return true;
```

Comment:
First of all - Are you at CakeFest :)?

While I could get away with that regarding Tree Behavior because it is hard to teach it composite foreign keys, the basic premise here is that the real foreign key is not just parent_id but [site_id, parent_id], e.g. site_id is both: a foreignKey and part of the composite primary key.

In other words:
In case you reference a foreign table where one field can be null and IS null, the foreign key reference is actually optional unless you specify all foreign key parts and then it HAS to exist.

Does the current `__fieldsAreNull()` check make sense to you?